src/passes/arrayLenFunctionalizer.ts

```diff
@@ -0,0 +1,34 @@
+import { FunctionStateMutability, generalizeType, getNodeType, MemberAccess } from 'solc-typed-ast';
+import { AST } from '../ast/ast';
+import { FunctionStubKind } from '../ast/cairoNodes';
+import { ASTMapper } from '../ast/mapper';
+import { createCairoFunctionStub, createCallToFunction } from '../utils/functionGeneration';
+import { isDynamicCallDataArray } from '../utils/nodeTypeProcessing';
+import { typeNameFromTypeNode } from '../utils/utils';
+
+export class ArrayLenFunctionalizer extends ASTMapper {
```

Comment:
I think this should be in the References subpass ArrayFunctions unless there's a good reason to keep it separate